Mathlib/GroupTheory/GroupAction/Basic.lean

```diff
@@ -106,6 +106,16 @@ theorem orbit.coe_smul {a : α} {m : M} {a' : orbit M a} : ↑(m • a') = m •
 #align mul_action.orbit.coe_smul MulAction.orbit.coe_smul
 #align add_action.orbit.coe_vadd AddAction.orbit.coe_vadd
 
+@[to_additive]
+lemma orbit_submonoid_subset (S : Submonoid M) (a : α) : orbit S a ⊆ orbit M a := by
```

Comment:
I wonder if you can use https://leanprover-community.github.io/mathlib4_docs/Mathlib/GroupTheory/Submonoid/Basic.html#SubmonoidClass to avoid having both a submonoid version and a subgroup version?

Reply:
The `MulAction` instance (and a load of other related instances) only exists for `Subgroup` and `Submonoid`, not for `SubmonoidClass`, so you can't even state the lemma for `SubmonoidClass` at present.

Reply:
How hard would it be to generalize those? Or would this be best left to another PR?

Reply:
Maybe I'll just do it myself in a separate PR.

Reply:
I think most of the subobject / hom / equiv classes probably have lots of scope to generalize existing definitions and results (including removing duplicate definitions / results for specific types if possible, which hasn't consistently been done when such generalization was done in the past). I'd expect lots more for `SubmonoidClass` than just those instances, so separate PRs definitely seem better for such generalizations.